instrumentation/basic_logger.go

```diff
@@ -86,6 +86,29 @@ func Stringable(key string, value fmt.Stringer) *Field {
 	return &Field{Key: key, String: value.String(), Type: StringType}
 }
 
+func StringableSlice(key string, values interface{}) *Field {
```

Comment:
I suggest we replace it with something like this to avoid performance hit:

```
func StringableSlice(key string, values ...fmt.Stringer) *Field {
```

Reply:
i think the problem with it is that you cannot pass the slice in its original type, so a []*SignedTransactions cannot be passed to that function signature, this is because the slice itself ([]) does not implement string(), but the objects inside.. (i.e. you have no generics, everything must implicitly implement the interfaces)

```
func StringableSlice1(key string, values ...fmt.Stringer) *Field {
	builder := strings.Builder{}
	builder.WriteString("[")

	for i, v := range values {
		builder.WriteString(v.String())
		if i < len(values)-1 {
			builder.WriteString(",")
		}
	}

	builder.WriteString("]")
	return &Field{Key: key, String: builder.String(), Type: StringType}
}
```